decisions/DR006-Generic-string-type.md

```diff
@@ -0,0 +1,179 @@
+  functions for operating on UTF-8 strings stored within a
+  `std::string`.
+* IBM's [ICU](https://icu.unicode.org/) popular heavyweight C (unicode)
+  string library with custom types.
+* [CsString](https://github.com/copperspice/cs_string) C++ library
+  providing dedicated unicode-aware string types.
+* [LabText](https://github.com/meshula/LabText) C-native string library
+  with C++ wrappers.
+* [SDS](https://github.com/antirez/sds) Redis's C-native string library.
+
+
+## Options considered
+
+### Option 1
+
+`std::string_view`/`std::string` for constants/parameters.
+
+#### Pros
+
+- Little boilerplate required.
+- Straightforward manipulation by `std::string`-compatible library
+  functions.
+- Consistent with other VFX open source projects (see above).
+
+#### Cons
+
+- Ties us to a specific underlying class that cannot be extended.
+- Sensitive to ABI differences between compilers (e.g. GCC C++11 ABI
+  break, see above).
+- Sensitive to cross-platform implementation differences.
+
+Estimated cost: Small
+
+### Option 2
+
+Custom type with hidden implementation ([pimpl](https://en.cppreference.com/w/cpp/language/pimpl)).
+
+#### Pros
+
+- Compiler/library isolation.
+- Allows use of intrusive smart pointers (e.g.
+  `enable_shared_from_this`).
+- Can be extended with custom logic for validation, encoding, etc.
+- Implementation can be extended without breaking ABI compatibility of
+  hosts/plugins.
+- A third-party library can be used under-the-hood.
+
+#### Cons
+
+- Additional boilerplate required, e.g. converting to/from native types,
+  and custom Python pybind11 converters.
+- (Small) performance hit of additional pointer indirection to
+  implementation.
+
+Estimated cost: Medium
+
+### Option 3
+
+Exposing a third-party library.
+
+#### Pros
+
+- No reinventing the wheel for (common) string functions.
+- Consistent implementation across compilers.
+- A C-native library (with C++ wrappers) would avoid the need for roll
+  our own C bindings.
+
+#### Cons
+
+- Additional third-party dependency - especially problematic if exposed
+  as a core type required by adopters, and even worse if adopters are
+  already using a different (or customized) version of the library.
+- Custom Python converters required.
+
+Estimated cost: Medium
+
+## Outcome
+
+We've tentatively decided on Option 1.
+
+This option seems to reflect the prevailing string treatment in the
+community. It offers the path of least resistance for bindings, both
+to Python and C.
+
+Our assumption of UTF-8 throughout, and our lack of a need for complex
+string parsing and manipulation, mean that we don't expect to require
+functionality much beyond what `std::string` offers. Any additional
+functionality (e.g. tokenizing UTF-8 strings) can be handled as a
+(hidden) implementation detail, potentially using one of the
+several existing third-party libraries that operate on `std::string`.
+
+If we wish to change the implementation in the future, one way we can
```

Comment:
[consider] Adding something along the lines of...

```suggestion
We will take advantage of the current "alpha" state of the project to
further explore and battle-test the practical ramifications of this
choice. We will have time before an initial release to change the
implementation should significant issues arise. Even beyond that point,
one way we can
```

Reply:
Incorporated that text in 653af661